presto-main/src/test/java/com/facebook/presto/type/TestCharOperators.java

```diff
@@ -192,4 +198,21 @@ public void testIndeterminate()
         assertOperator(INDETERMINATE, "CAST(null AS CHAR(3))", BOOLEAN, true);
         assertOperator(INDETERMINATE, "CHAR '123'", BOOLEAN, false);
     }
+
+    @Test
+    public void testCharCast()
```

Comment:
Can you add some test cases for the following:

1) some numbers without whitespace
2) What happens when the number is outside of the range of values for integer/float
3) Any support for nan or infinity
4) any support for hexidecimal numbers
5) Any support for hexidecimal floating point numbers

Reply:
Added some more tests. There is no support for hexa decimal numbers now.

Reply:
We could probably add support for hexadecimals by considering any char(x) type prefixed with 0x as a hexadecimal, but I'm not sure if that will be an unintended output for those who use the cast function